src/io_uring.c

```diff
@@ -0,0 +1,97 @@
+#include "io_uring.h"
+
+#ifdef HAVE_LIBURING
+#include <liburing.h>
+#include <string.h>
+
+#include "serverassert.h"
+#include "zmalloc.h"
+
+#define IO_URING_DEPTH 256 /* io_uring instance queue depth. */
+
+static size_t io_uring_queue_len = 0;
+static io_uring *_io_uring;
+
+void initIOUring(void) {
```

Comment:
Can we let the caller worried about the life cycle of the io_uring? Then we don't need `getIOUring` and we get to save it to the server struct :-)

```suggestion
io_uring* createIOUring(void) {
```

Reply:
Done.

Reply:
Ping, just to be clear, what's your plan here?


Shall we pass the `io_uring` pointer to all function in this file, so this file doesn't need to include `server.h`, but we still store the `io_uring` in the `server` struct?

Reply:
>Ping, just to be clear, what's your plan here?

To achieve minimum coupling and maximum cohesiveness.